tdrs-backend/tdpservice/parsers/util.py

```diff
@@ -11,20 +13,60 @@ def value_is_empty(value, length):
     return value is None or value in empty_values
 
 
+def generate_parser_error(datafile, line_number, schema, error_category, error_message, record=None, field=None):
+    """Create and return a ParserError using args."""
+    model = schema.model if schema else None
+
+    # make fields optional
+    return ParserError.objects.create(
+        file=datafile,
+        row_number=line_number,
+        column_number=getattr(field, 'item', 0),
+        item_number=getattr(field, 'item', 0),
+        field_name=getattr(field, 'name', 'none'),
+        category=error_category,
+        case_number=getattr(record, 'CASE_NUMBER', None),
+        error_message=error_message,
+        error_type=error_category,
+        content_type=ContentType.objects.get(
+            model=model if record and not isinstance(record, dict) else 'ssp_m1'
```

Comment:
I may have asked this before, what's the significant of `else 'ssp_m1'`?

Reply:
this was an oversight on my part. i had included default values for fields that should be nullable, i went ahead and made the required model changes and changed the defaults to `None`s